packages/shell-api/src/top-level-api.ts

```diff
@@ -0,0 +1,72 @@
+import ShellApi from './shell-api';
+
+class TopLevelApi extends ShellApi {
+  /**
+   * Returns true if a value is a shell api type
+   *
+   * @param {any} evaluationResult - The result of evaluation
+   */
+  private isShellApiType(evaluationResult: any): boolean {
+    return evaluationResult &&
+      typeof evaluationResult.shellApiType === 'function' &&
+      typeof evaluationResult.toReplString === 'function';
+  }
+
+  /**
+   * Checks for linux-style commands then evaluates input using originalEval.
+   *
+   * @param {function} originalEval - the javascript evaluator.
+   * @param {String} input - user input.
+   * @param {Context} context - the execution context.
+   * @param {String} filename
+   */
+  private async innerEval(originalEval: any, input: string, context: any, filename: string) {
+    const argv = input.trim().split(' ');
+    const cmd = argv[0];
+    argv.shift();
+    switch (cmd) {
+      case 'use':
+        return this.use(argv[0]);
+      case 'show':
+        return this.show(argv[0]);
+      case 'it':
+        return this.it();
+      case 'help':
+        return this.help();
+      case 'var':
+        this._mapper.cursorAssigned = true;
```

Comment:
always wanted to ask this one, do we have to do something more here? like handle also `let`, `const`, direct assignment? Could we just reset the cursor?

Reply:
I wish we could remove that case but not iterating the cursor when it's not returned to the evaluator is expected current shell behavior that we're not going to be able to remove anytime soon I think. I do think you're right that we should do some more checking to see if cursor is assigned or not instead of just parsing for `var`. So get rid of `cursorAssigned` entirely, assume that all cursors are not iterated. It would probably happen as you said in the print statement, where we check if the value returned to the evaulator is of type cursor and if so then iterate it.

Otherwise we would have to pass some state into the async rewriter so that we could record when parsing if the find command result is assigned to anything. I made a ticket: https://jira.mongodb.org/browse/MONGOSH-161